cli/lsp/tsc.rs

```diff
@@ -411,6 +413,77 @@ impl QuickInfo {
   }
 }
 
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RenameLocation {
+  // inherit from DocumentSpan
+  text_span: TextSpan,
+  file_name: String,
+  original_text_span: Option<TextSpan>,
+  original_file_name: Option<String>,
+  context_span: Option<TextSpan>,
+  original_context_span: Option<TextSpan>,
+  // RenameLocation props
+  prefix_text: Option<String>,
+  suffix_text: Option<String>,
+}
+
+pub struct RenameLocations {
+  pub locations: Vec<RenameLocation>,
+}
+
+impl RenameLocations {
+  pub async fn into_workspace_edit(
+    self,
+    ls: &LanguageServer,
+    new_name: &str,
+  ) -> lsp_types::WorkspaceEdit {
+    let mut text_document_edit_map: HashMap<Url, lsp_types::TextDocumentEdit> =
+      HashMap::new();
+    for location in self.locations.iter() {
+      // ensure TextDocumentEdit for `location.file_name`.
+      let uri = utils::normalize_file_name(&location.file_name).unwrap();
+      if text_document_edit_map.get(&uri).is_none() {
+        text_document_edit_map.insert(
+          uri.clone(),
+          lsp_types::TextDocumentEdit {
+            text_document: lsp_types::OptionalVersionedTextDocumentIdentifier {
+              uri: uri.clone(),
+              version: None,
```

Comment:
I am not sure, but shouldn't this be the version of the text document the edit happened on? cc @kitsonk

Reply:
Correct, to ensure there aren't "race" conditions in the editor.

Reply:
Thank you for your reviews.

I've added a version from `StateSnapshot`.